### ISSUE TITLE
no-unused-vars rule flags a variable that is used

### ISSUE THREAD

**Issue by stevenlyons** [eslint/eslint#12830]
**Tell us about your environment**

Node version: v13.6.0
npm version: v6.13.4
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
  },
  "rules": {
    "semi": [2, "always"],
    "camelcase": [0, { "properties": "never" }],
    "newline-after-var": [2, "always"],
    "no-unused-vars": "error"
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
    it('eslint reproduction', function () {
      let interval = 1000;
      let current = 0;

      // Progress the interval amount
      current = advancePosition(interval, current);

      // Do stuff...

      // Progress again the interval amount
      current = advancePosition(interval, current);

      // Do more stuff...
    });
```

```bash
eslint 'test/**/*.js' --ignore-pattern 'lib/vendor/**/*.js' --quiet",
```

**What did you expect to happen?**
I expected this function to parse without errors since the variable is used.

**What actually happened? Please include the actual, raw output from ESLint.**

```156:11  error  'current' is assigned a value but never used       no-unused-vars```

The ```current``` variable was flagged as unused but it was actually used for keeping track of the current position.

I am upgrading the project from a much earlier version of eslint (2.13.1) to the latest release version. This instance was not flagged as an error in the previous version. 

**Are you willing to submit a pull request to fix this bug?**

If I was able to fix this issue, I would be willing to, yes.

**Comment by yeonjuan** [comment]
@stevenlyons 
Hi! Thanks for the report! :)

> A variable is not considered to be used if it is only ever declared (var foo = 5) or assigned to (foo = 7).

By the [docs in no-unused-vars](https://eslint.org/docs/rules/no-unused-vars#rule-details), this seems to be intended behavior because the variable `current` is assigned with value but not used for anything.

**Comment by stevenlyons** [comment]
Thanks for the quick reply @yeonjuan. I assumed it should be counted as used because it meets the the following criteria from the docs you linked to: "It is passed into a function as an argument (doSomething(foo))". Or I may be mis-interepting how to read that criteria?

**Comment by mdjermanovic** [comment]
I believe this is related to the change #6535 (issue #6348) and the rule works like that since `v3.0.0`.

In short, `current = advancePosition(interval, current)` doesn't count as being used, because it's used to update itself.

A solution could be to replace the last `current = advancePosition(interval, current);` with just `advancePosition(interval, current)`.

**Comment by stevenlyons** [comment]
Thanks for the info, appreciate it.
